Sources/Basic/misc.swift

```diff
@@ -57,20 +60,33 @@ public func getEnvSearchPaths(
 ///
 /// - Parameters:
 ///   - filename: The name of the file to find.
-///   - cwd: The current working directory to look in.
+///   - currentWorkingDirectory: The current working directory to look in.
 ///   - searchPaths: The additional search paths to look in if not found in cwd.
 /// - Returns: Valid path to executable if present, otherwise nil.
 public func lookupExecutablePath(
     filename value: String?,
-    currentWorkingDirectory cwd: AbsolutePath = currentWorkingDirectory,
+    currentWorkingDirectory: AbsolutePath? = localFileSystem.currentWorkingDirectory,
     searchPaths: [AbsolutePath] = []
 ) -> AbsolutePath? {
+
     // We should have a value to continue.
     guard let value = value, !value.isEmpty else {
         return nil
     }
-    // We have a value, but it could be an absolute or a relative path.
-    let path = AbsolutePath(value, relativeTo: cwd)
+
+    let path: AbsolutePath
+    if let cwd = currentWorkingDirectory {
+        // We have a value, but it could be an absolute or a relative path.
+        path = AbsolutePath(value, relativeTo: cwd)
+    } else {
+        // Current directory not being available is not a problem
+        // for the absolute-specified paths.
+        guard value.hasPrefix("/") else {
+            return nil
+        }
+        path = AbsolutePath(value)
```

Comment:
This might be more clear:

```swift
if let cwd = currentWorkingDirectory {
    path = AbsolutePath(value, relativeTo: cwd)
} else if let absPath = try? AbsolutePath(validating: value) {
    path = absPath
} else {
    return nil
}
```